### PR TITLE
Make PRam interactive

### PR DESCRIPTION
**Pull request by Polynomial-C**

### CODE REVIEW

File: pram [hunk 1]
```diff
@@ -25,6 +25,8 @@ print_help() {
 	echo "                   a bug number or URL, can be specified multiple times"
 	echo "  -e EDITOR, --editor EDITOR"
 	echo "                   Override the editor used (default: \${EDITOR})"
+	echo "  -n, --non-interactive"
```
Comment (mgorny) [inline pram]: Make it `-I`, please.

File: pram [hunk 2]
```diff
@@ -25,6 +25,8 @@ print_help() {
...
+	echo "                   Do not interactively ask to merge, just do it."
```
Comment (mgorny) [inline pram]: No dot elsewhere, no dot here ;-).

File: pram [hunk 3]
```diff
@@ -55,6 +57,7 @@ main() {
 	local prs=()
 	local repo=gentoo/gentoo
 	local signoff=1
+	local interactive=true
```
Comment (mgorny) [inline pram]: Please keep the coding style, i.e. `=1`.

File: pram [hunk 4]
```diff
@@ -100,6 +103,9 @@ main() {
 			--editor=*)
 				editor=${1#*=}
 				;;
+			-n|--non-interactive)
+				interactive=false
```
Comment (mgorny) [inline pram]: `=` (empty)

File: pram [hunk 5]
```diff
@@ -218,6 +224,19 @@ main() {
 			continue
 		fi
 
+		if ${interactive} ; then
```
Comment (mgorny) [inline pram]: Don't do such horrible things as unnecessary var→command magic.

File: pram [hunk 6]
```diff
@@ -218,6 +224,19 @@ main() {
...
+			local answer
+			read -p "Do you want to merge this? (y/n) " answer
+			case ${answer} in
```
Comment (mgorny) [inline pram]: `${answer,,}` to make it case-insensitive.

File: pram [hunk 7]
```diff
@@ -218,6 +224,19 @@ main() {
...
+				y)
+					:;
```
Comment (mgorny) [inline pram]: Drop this.

File: pram [hunk 8]
```diff
@@ -218,6 +224,19 @@ main() {
 			continue
 		fi
 
+		if ${interactive} ; then
+			local answer
+			read -p "Do you want to merge this? (y/n) " answer
+			case ${answer} in
+				y)
```
Comment (mgorny) [inline pram]: How about allowing empty string too? I suppose we should default to yes. Also make it `(Y/n)` in the prompt then.

File: pram [hunk 9]
```diff
@@ -218,6 +224,19 @@ main() {
...
+					:;
+					;;
+				*)
+					echo "Merge has been skipped!"
+					return 0
```
Comment (mgorny) [inline pram:234]: You are omitting remaining parameters here. If you really want to do that, please add extra `/q` for quit.

**Review by mgorny — CHANGES_REQUESTED** [review]

File: pram (hunk 7 above)
Comment (Polynomial-C) [reply pram]: I can't drop this or else this happens:
```
$ LANG="C" ~/bin/pram 11429                                  
/home/polynomial-c/bin/pram: line 234: syntax error near unexpected token `)'
/home/polynomial-c/bin/pram: line 234: `                                *)
```

Comment (Polynomial-C) [reply pram]: Nevermind, I removed the wrong line.

File: pram (hunk 9 above)
Comment (Polynomial-C) [reply pram:234]: I do not really understand what you mean here. Care to explain a bit more?

Comment (Polynomial-C) [reply pram:234]: Would changing it to `continue` be sufficient?

Comment (mgorny) [reply pram:234]: Yes, probably. Just make sure it doesn't skip something necessary below.

I meant that you can pass multiple PRs to it, and you don't want 'n' to mean 'skip all remaining PRs'. I suppose we'd want two options: 'n' to skip current, 'q' to quit (= skip all).

Comment (Polynomial-C) [reply pram:234]: How about this?

**Review by mgorny — APPROVED** [review]
LGTM

**Pull request closed without merging**